models/bundle.go

```diff
@@ -110,6 +111,20 @@ func (x *BundleEntryComponent) UnmarshalJSON(data []byte) (err error) {
 	return
 }
 
+// Custom SetBSON implementation to properly deserialize embedded resources
+// otherwise represented as interface{} into resource-specific structs as they
+// are retrieved from the database.
+func (x *BundleEntryComponent) SetBSON(raw bson.Raw) (err error) {
```

Comment:
This is good.  It would be even _better_ if there was an accompanying unit test!